app/api/odm/model.ts

```diff
@@ -28,33 +46,48 @@ class UpdateLogHelper {
     );
   }
 
-  async upsertLogMany(affectedIds: any[], deleted = false) {
-    await updatelogsModel._updateMany(
-      { mongoId: { $in: affectedIds }, namespace: this.collectionName },
-      { $set: { timestamp: Date.now(), deleted } },
-      { lean: true }
-    );
+  async upsertLogMany(
+    query: UwaziFilterQuery<T>,
+    deleted = false,
+    batchSize = UpdateLogHelper.batchSizeUpsertMany
+  ) {
+    await new PromisePool()
+      .for(await getBatchSteps(models[this.collectionName], query, batchSize))
+      .withConcurrency(5)
+      .process(async (stepIndex: T) => {
+        const batch = await models[this.collectionName].get(
+          { ...query, $and: [{ _id: { $gte: stepIndex } }] },
+          { _id: 1 },
+          { limit: batchSize }
```

Comment:
I have a couple of questions here:
- since we have all the ids already, is it more performant to use the limit than it would be to do a `@gte` AND `@lt` ?
- why is this $and required here?  is it not just "id greater than stepIndex"?  Is it because there could be also an _id key in the query already? (probably the case, just confirming)

Reply:
> * since we have all the ids already, is it more performant to use the limit than it would be to do a `@gte` AND `@lt` ?


I dont think i understand the question, i do not think limit has a performance overhead, normally this problem appears using offset and limit.


> * why is this $and required here?  is it not just "id greater than stepIndex"?  Is it because there could be also an _id key in the query already? (probably the case, just confirming)


yes this is the case, actually it was the case in the test itself, that is why i had to add the $and 🙈